File: app/src/main/java/com/fengniao/news/net/OkHttpManager.java
package com.fengniao.news.net;

import okhttp3.Call;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;

/**
 * Created by a1 on 2017/4/12.
 */

public class OkHttpManager {
    private static OkHttpManager mInstance;
    private OkHttpClient mOkHttpClient;


    private OkHttpManager() {
        mOkHttpClient = new OkHttpClient();
    }

    public static OkHttpManager getInstance() {
        if (mInstance == null) {
            mInstance = new OkHttpManager();
        }
        return mInstance;
    }


    //异步get
    public void getAsyn(String url, FNCallbackHolder holder) {
        Request request = new Request.Builder().url(url).build();
        Call call = mOkHttpClient.newCall(request);
        call.enqueue(holder);
    }

    //异步post
    public void postAsyn(String url, RequestBody body, FNCallbackHolder holder) {
        Request request = new Request.Builder().url(url).post(body).build();
        Call call = mOkHttpClient.newCall(request);
        call.enqueue(holder);
    }
}

File: app/src/main/java/com/fengniao/news/net/OkHttpManager.kt
package com.fengniao.news.net

import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody

/**
 * Created by a1 on 2017/4/12.
 */

class OkHttpManager private constructor() {
    private val mOkHttpClient: OkHttpClient = OkHttpClient()


    //异步get
    fun getAsyn(url: String, holder: FNCallbackHolder) {
        val request = Request.Builder().url(url).build()
        val call = mOkHttpClient.newCall(request)
        call.enqueue(holder)
    }

    //异步post
    fun postAsyn(url: String, body: RequestBody, holder: FNCallbackHolder) {
        val request = Request.Builder().url(url).post(body).build()
        val call = mOkHttpClient.newCall(request)
        call.enqueue(holder)
    }

    companion object {
        private var mInstance: OkHttpManager? = null

        val instance: OkHttpManager
            get() {
                if (mInstance == null) {
                    mInstance = OkHttpManager()
                }
                return mInstance as OkHttpManager
            }
    }
}
